samples/java-customer-registry/src/main/java/customer/Main.java

```diff
@@ -37,6 +53,24 @@ public static void main(String[] args) throws Exception {
               "customerByName",
               CustomerDomain.getDescriptor())
           // end::register[]
+          .lowLevel()
+          .registerValueEntity(
+              // TODO this factory probably can disappear, we can pass in the handler directly?
+              context ->
+                  new CustomerValueEntityHandler(
+                      // TODO I guess construction the Entity should somehow remain part of the
+                      // customer API, so they can pass
+                      // in  anything they like in the constructor, including the context?
```

Comment:
yes, we should change the registration methods to take a factory lambda `CreationContext -> ValueEntity`